jvm-packages/python/sparkxgb/util.py

```diff
@@ -0,0 +1,68 @@
+#
+# Copyright (c) 2018 by Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from pyspark.ml.util import JavaMLReadable, JavaMLReader
+
+
+class XGBoostReadable(JavaMLReadable):
+    """
+    Mixin for instances that provide XGBoostReader.
+    """
+
+    @classmethod
+    def read(cls):
+        """Returns an XGBoostReader instance for this class."""
+        return XGBoostReader(cls)
+
+
+class XGBoostReader(JavaMLReader):
+    """
+    A reader for XGBoost objects.
+    """
+
+    def __init__(self, clazz):
+        self._clazz = clazz
+        self._jread = self._load_java_obj(clazz).read()
+
+    @classmethod
+    def _java_loader_class(cls, clazz):
+        """
+        Returns the full class name of the Java XGBoost class. In this case,
+        we prepend "ml.dmlc.xgboost4j.scala.spark" to the name of the class.
+        """
+
+        # Deal with Python locations for XGBoost objects.
+        if clazz.__name__ in ("XGBoostEstimator", "XGBoostClassificationModel", "XGBoostRegressionModel"):
+            java_package = clazz.__module__.replace("sparkxgb", "ml.dmlc.xgboost4j.scala.spark")
+
+        # Our Pipeline object calls down to Spark's Pipeline object.
```

Comment:
Is it really necessary to separate each else branch with a comment? Or just give our logic note ?